packages/compass-components/src/components/workspace-container.tsx

```diff
@@ -1,15 +1,70 @@
-import React from 'react';
+import React, { useRef } from 'react';
 import { css, cx } from '@leafygreen-ui/emotion';
 import { uiColors } from '@leafygreen-ui/palette';
-
-import { withTheme } from '../hooks/use-theme';
 import { gray8 } from '../compass-ui-colors';
+import { Theme, useTheme, withTheme } from '../hooks/use-theme';
+import { spacing } from '@leafygreen-ui/tokens';
+import { transparentize } from 'polished';
+import { useInView } from 'react-intersection-observer';
 
 const workspaceContainerStyles = css({
   height: '100%',
   width: '100%',
   display: 'flex',
+  flexDirection: 'column',
+  overflow: 'hidden',
+});
+
+const toolbarStyles = css({
+  flex: 'none',
+});
+
+const scrollBoxStyles = css({
+  flex: 1,
+  width: '100%',
+  minHeight: 0,
+  position: 'relative',
+  zIndex: 0,
+});
+
+const shadowHeight = spacing[1];
+
+const shadowStyles = css({
+  position: 'absolute',
+  top: 0,
+  display: 'block',
+  width: '100%',
+  height: shadowHeight,
+  borderTop: `1px solid`,
+  flex: 'none',
+  zIndex: 1,
```

Comment:
Even if the overlap is small, I'd suggests disable pointer events on the shadow to make sure it doesn't mess with clicking

```suggestion
  zIndex: 1,
  pointerEvents: 'none',
```

Reply:
Ah, nice idea! didn't know this let the clicks through